gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/AndroidPluginIntegration.kt

```diff
@@ -62,6 +63,19 @@ object AndroidPluginIntegration {
             .map { it.name }
     }
 
+    @Suppress("DEPRECATION")
+    fun updateKspWithAndroidSourceSets(
+        kotlinCompilation: KotlinJvmAndroidCompilation,
+        kspTaskProvider: TaskProvider<out KspTaskJvm>
+    ) {
+        kotlinCompilation.androidVariant.getSourceFolders(SourceKind.JAVA).forEach { source ->
```

Comment:
Do you know whether this includes KSP generated code or not? E.g., does this include anything under `workload/build/generated/ksp`?

Reply:
KSP registers its generated code to the default source set of each compilation target. I wonder how AGP gets the sources and in particular whether `kotlinCompilation.androidVariant.getSourceFolders(SourceKind.JAVA)` returns the content of the default source set. If the `getSourceFolders` only returns sources that are registered through AGP's API, it should be fine. If it includes anything from the default source set, we may need to exclude KSP's generated code explicitly; Otherwise otherwise be a dependency loop or the code that is generated in previous build get incorrectly picked up for processing.

Reply:
Assuming that `default source set` is src/main folder that is shared across variants. getSourceFolders does return default source set as far as I remember (along with 5 other folders like variant specific source folder, aidl, viewBinding, BuildConig generated). Still, ksp registers folder with generated sources separately for each android variant - debug, release. And sources for each variant are located in separate `generated/<variant name>/*` folder. This way we are isolated within one variant. And as long as we adding ksp generated folder to android in last function, there is no risk variant generated folder will be part of what initially `getSourceFolders` returns 

 I saw circular dependency issue earlier when registerGeneratedSources was called before tryUpdateKspWithAndroidSourceSets (I fixed this situation).  Circuit was with ksp generated source set because it was set as something that ksp depends  on and that ksp task produces